src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -249,6 +251,21 @@
     SingleIdentifierGrammar=ansi_dialect.get_grammar("SingleIdentifierGrammar").copy(
         insert=[Ref("SessionVariableNameSegment")]
     ),
+    AndKeywordSegment=OneOf(
+        StringParser("AND", KeywordSegment, type="binary_operator"),
+        StringParser(
+            "&&", CodeSegment, name="double_ampersand", type="binary_operator"
+        ),
+    ),
+    OrKeywordSegment=OneOf(
+        StringParser("OR", KeywordSegment, type="binary_operator"),
+        StringParser("||", CodeSegment, name="double_pipe", type="binary_operator"),
+        StringParser("XOR", KeywordSegment, type="binary_operator"),
+    ),
+    NotKeywordSegment=OneOf(
+        StringParser("not", KeywordSegment, type="keyword"),
+        StringParser("!", KeywordSegment, type="keyword"),
```

Comment:
I wouldn't call this a keyword:

```suggestion
        StringParser("!", CodeSegment, type="not_operator"),
```

Reply:
Great. (I actually called it "unary_operator" in my first version.)

Reply:
Happy with `unary_operator`. Was trying to think of the right but couldn't!